src/Nuget.PackageIndex.VisualStudio/DnxProjectFilter.cs

```diff
@@ -0,0 +1,192 @@
+                    {
+                        return;
+                    }
+
+                    projectFullName = project.FullName.ToLowerInvariant();
+                }
+                catch (Exception e)
+                {
+                    // we should not throw here, since it would create an exception that may be 
+                    // visible to the user, instead just dump into debugger output or to package
+                    // manager console.
+                    // TODO Package manager console?
+                    Debug.Write(e.ToString());
+                }
+            });
+
+            if (!string.IsNullOrEmpty(projectFullName))
+            {
+                FilesCache.AddOrUpdate(filePath, (k) => projectFullName, (k, v) => projectFullName);
+            }
+
+            return IsProjectFileSupported(projectFullName);
+        }
+
+        private bool IsProjectFileSupported(string projectFile)
+        {
+            return !string.IsNullOrEmpty(projectFile) && projectFile.EndsWith(".xproj");
+        }
+
+        private void ClearCache(string projectFullPath)
+        {
+            if (string.IsNullOrEmpty(projectFullPath))
+            {
+                return;
+            }
+
+            // remove files that belong to the project that was removed/renamed
+            projectFullPath = projectFullPath.ToLowerInvariant();
+            foreach (var kvp in FilesCache.ToList())
+            {
+                // project paths in the cache arready in lower case
+                if (kvp.Value.Equals(projectFullPath))
+                {
+                    string removedVal = string.Empty;
+                    FilesCache.TryRemove(kvp.Key, out removedVal);
+                }
+            }
+        }
+
+        private void OnBeforeSolutionClosed()
+        {
+            _solutionIsClosing = true;
+        }
+
+        private void OnSolutionOpened()
+        {
+            _solutionIsClosing = false;
+        }
+
+        private void OnProjectRemoved(Project project)
+        {
+            Debug.Assert(project != null, "Project passed to OnProjectRemoved DTE event was null.");
+            if (project == null)
+            {
+                return;
+            }
+
+            var projectFullPath = string.Empty;
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                // Switch to main thread
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                projectFullPath = project.FullName;
+            });
+
+            ClearCache(projectFullPath);
+        }
+
+        private void OnProjectRenamed(Project project, string oldName)
+        {
+            ClearCache(oldName);
+        }
+
+        #region IDisposable
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
```

Comment:
the private method is unnecessary. The dispose(true) pattern is used for derived classes.

Also avoid `#region` in the code